### PR TITLE
Complete home search and add Zone page.

### PR DESCRIPTION
**Pull request by guavajellyaaron**
Added new repo method to get Bosses by zone
Added new service method to get clean Zone url
Updated search method to pull in game and zone names
Added zone page that displays list of bosses

### CODE REVIEW

File: projects/BossCheatSheets.Web/App_Start/RouteConfig.cs [hunk 1]
```diff
@@ -45,6 +46,18 @@ public static void RegisterRoutes(RouteCollection routes)
                 constraints: new { id = @"\d+" }
             );
 
+            //world_of_warcraft/hellfire_citadel/fel_lord_zakuun/Another_strat/3
+            routes.MapRoute(
```
Comment (guavajellyaaron) [inline projects/BossCheatSheets.Web/App_Start/RouteConfig.cs]: hmm. I think I created a bug here. It matches too closely to the default and picks that up instead.
For example, 
http://bosscheatsheets.web/Strategy/Create/2
Redirects to 
http://bosscheatsheets.web/world_of_warcraft/gate_of_the_setting_sun/2

What is the best way to fix this?

File: projects/BossCheatSheets.DomainModel/Services/IStrategyService.cs [hunk 2]
```diff
@@ -5,5 +5,6 @@ public interface IStrategyService
         string GetStrategyURL(int strategyID);
         string VerifyBossWithSpecURL(string gameName, string zoneName, string bossName, int id, string spec, int specId);
         string VerifyBossURL(string gameName, string zoneName, string bossName, int id);
+        string VerifyZoneURL(string gameName, string zoneName, int id);
     }
 }
```
Comment (happygaijin) [inline projects/BossCheatSheets.DomainModel/Services/IStrategyService.cs]: This is looking more and more like a URLVerification service :)

File: projects/BossCheatSheets.Web/App_Start/RouteConfig.cs (hunk 1 above)
Comment (happygaijin) [inline projects/BossCheatSheets.Web/App_Start/RouteConfig.cs]: http://bosscheatsheets.web/Strategy/Create/{id} <-- Strategy and Create seem like two words that will always be in that URL yes? So just create another route with those two words hardcoded and put it before the more generic one.

File: projects/BossCheatSheets.Web/Controllers/HomeController.cs [hunk 3]
```diff
@@ -47,9 +47,9 @@ public ActionResult StrategyDetail(int id)
         [HttpGet]
         public ActionResult Search(string q)
         {
-            string dbQuery = "SELECT 'Boss' AS Type, BossID AS ID, Name AS Name FROM Boss WHERE Name LIKE '%" + q + "%' " +
+            string dbQuery = "SELECT 'Boss' AS Type, BossID AS ID, Boss.Name AS Title, Zone.Name AS ZoneName, Game.Name AS GameName FROM Boss RIGHT JOIN Zone ON Boss.ZoneID = Zone.ZoneID RIGHT JOIN Game ON Zone.GameID = Game.GameID WHERE Boss.Name LIKE '%" + q + "%' " +
                              "UNION ALL " +
-                             "SELECT 'Zone' AS Type, ZoneID AS ID, Name AS Name FROM Zone WHERE Name LIKE '%" + q + "%' ";
+                             "SELECT 'Zone' AS Type, ZoneID AS ID, Zone.Name AS Title, Zone.Name AS ZoneName, Game.Name AS GameName FROM Zone RIGHT JOIN Game ON Zone.GameID = Game.GameID WHERE Zone.Name LIKE '%" + q + "%' ";
```
Comment (happygaijin) [inline projects/BossCheatSheets.Web/Controllers/HomeController.cs]: We should talk about what you want to do here. Maybe it's right, but I can't tell what you want to do with this query exactly. I mean I know what search is supposed to do, I just want to track with your way of tackling it.

**Pull request merged**